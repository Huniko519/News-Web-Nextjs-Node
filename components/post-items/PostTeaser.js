import React from 'react';
import PropTypes from 'prop-types';
import PostItemBadge from './PostItemBadge';
import PostItemMedia from './PostItemMedia';
import PostItemTitle from './PostItemTitle';
import PostItemLabel from './PostItemLabel';
import PostItemPubDate from './PostItemPubDate';
import Qoute from '../../utils/Qoute';

/**
 * Post Teasers feature a full(component)width image stacked on top of the post headline and primary
 * category.
 *
 * ╔══════════╗
 * ║          ║
 * ║  Image   ║
 * ║          ║
 * ╟──────────╢
 * ║ Category ║
 * ║          ║
 * ║ Headline ║
 * ╚══════════╝
 *
 * @param {post: {...postData}} props
 */
const PostTeaser = (props) => {
  const { post, showAuthor, showPubDate } = props;

  return (
    <div
      className={`inews__post inews__post-teaser ${showAuthor ? 'show-author' : ''} ${post.template ? post.template : ''}`}
      key={post.id}
      data-post-id={post.id}
    >

      <PostItemMedia
        className="inews__post-teaser__media"
        imageSize={!showAuthor ? 'medium-16:9' : 'small-1:1'}
        width={!showAuthor ? 16 : 155}
        height={!showAuthor ? 9 : 155}
        layout={!showAuthor ? 'responsive' : 'fixed'}
        showAuthor={showAuthor}
        postItemType="teaser"
        placeholder={`/static/images/placeholder/${!showAuthor ? 'placeholder-640x360.png' : 'placeholder-84x84.png'}`}
        post={post}
      />
      <div className="inews__post-teaser__content">
        <div>
          <PostItemBadge post={post} showAuthor={showAuthor || post.template === 'opinion'} />
          <span className="inews__post-teaser__content__headline">
            {post.template === 'opinion' && <Qoute size="small" />}
            <PostItemLabel post={post} />
            <PostItemTitle post={post} />
          </span>
        </div>
        <PostItemPubDate post={post} showPubDate={showPubDate} />
      </div>
    </div>
  );
};

/* istanbul ignore next */
PostTeaser.propTypes = {
  post: PropTypes.objectOf(PropTypes.any).isRequired,
  showAuthor: PropTypes.bool,
  showPubDate: PropTypes.bool,
};

/* istanbul ignore next */
PostTeaser.defaultProps = {
  showAuthor: false,
  showPubDate: false,
};

export default PostTeaser;
