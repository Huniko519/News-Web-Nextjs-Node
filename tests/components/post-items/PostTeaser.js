import React from 'react';
import PostTeaser from '../../../components/post-items/PostTeaser';

test('renders the snapshot', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postTeaser = shallow(<PostTeaser post={post} />);
  expect(postTeaser).toMatchSnapshot();
});

it('should output with the correct class on the root element', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postTeaser = shallow(<PostTeaser post={post} />);
  expect(postTeaser.hasClass('inews__post')).toBe(true);
  expect(postTeaser.hasClass('inews__post-teaser')).toBe(true);
});

it('should call the <PostItemMedia /> component with the correct props', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postTeaser = shallow(<PostTeaser post={post} />);
  const postItemMedia = postTeaser.find('PostItemMedia');
  expect(postItemMedia.length).toBe(1);
  expect(postItemMedia.prop('className')).toBe('inews__post-teaser__media');
  expect(postItemMedia.prop('imageSize')).toBe('medium-16:9');
  expect(postItemMedia.prop('width')).toBe(16);
  expect(postItemMedia.prop('height')).toBe(9);
  expect(postItemMedia.prop('layout')).toBe('responsive');
  expect(postItemMedia.prop('placeholder')).toBe('/static/images/placeholder/placeholder-640x360.png');
  expect(postItemMedia.prop('post')).toMatchObject(post);
});

it('should call the <PostItemTitle /> component with the correct props', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postTeaser = shallow(<PostTeaser post={post} />);
  const postItemMedia = postTeaser.find('PostItemTitle');
  expect(postItemMedia.length).toBe(1);
  expect(postItemMedia.prop('post')).toMatchObject(post);
});

it('should output a <PostItemBadge /> component', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postTeaser = shallow(<PostTeaser post={post} />);
  expect(postTeaser.find('PostItemBadge').length).toBe(1);
  expect(postTeaser.find('PostItemBadge').prop('post')).toMatchObject(post);
});
