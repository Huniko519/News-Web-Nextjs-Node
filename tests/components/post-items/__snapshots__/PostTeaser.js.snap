// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the snapshot 1`] = `
<div
  className="inews__post inews__post-teaser  "
  data-post-id={1}
  key="1"
>
  <PostItemMedia
    className="inews__post-teaser__media"
    height={9}
    imageSize="medium-16:9"
    layout="responsive"
    linkToAuthor={false}
    placeholder="/static/images/placeholder/placeholder-640x360.png"
    post={
      Object {
        "breadcrumbs": Array [
          Object {
            "name": "News",
          },
        ],
        "id": 1,
        "link": "https://inews-article.com/test-article",
        "title": Object {
          "rendered": "title",
        },
      }
    }
    postItemType="teaser"
    showAuthor={false}
    width={16}
  />
  <div
    className="inews__post-teaser__content"
  >
    <div>
      <PostItemBadge
        post={
          Object {
            "breadcrumbs": Array [
              Object {
                "name": "News",
              },
            ],
            "id": 1,
            "link": "https://inews-article.com/test-article",
            "title": Object {
              "rendered": "title",
            },
          }
        }
        showAuthor={false}
      />
      <span
        className="inews__post-teaser__content__headline"
      >
        <PostItemLabel
          post={
            Object {
              "breadcrumbs": Array [
                Object {
                  "name": "News",
                },
              ],
              "id": 1,
              "link": "https://inews-article.com/test-article",
              "title": Object {
                "rendered": "title",
              },
            }
          }
        />
        <PostItemTitle
          post={
            Object {
              "breadcrumbs": Array [
                Object {
                  "name": "News",
                },
              ],
              "id": 1,
              "link": "https://inews-article.com/test-article",
              "title": Object {
                "rendered": "title",
              },
            }
          }
        />
      </span>
    </div>
    <PostItemPubDate
      post={
        Object {
          "breadcrumbs": Array [
            Object {
              "name": "News",
            },
          ],
          "id": 1,
          "link": "https://inews-article.com/test-article",
          "title": Object {
            "rendered": "title",
          },
        }
      }
      showPubDate={false}
    />
  </div>
</div>
`;
